codice/ddf
# DDF-4317 As a user, I want to copy MGRS and UTM/UPS coordinates from the map to the clipboard

#### What does this PR do?
Adds support for the user to right click the map and copy the currently selected coordinate string to the clipboard in MGRS and UTM/UPS formats. 

#### Who is reviewing it? 
@Variadicism 
@samuelechu 
@BernardIgiri 
@Dimitry-Sherman

#### Select relevant component teams: 
@codice/ui 

#### Ask 2 committers to review/merge the PR and tag them here.
@bdeining 
@djblue

#### How should this be tested?
Right click around the maps and ensure all formats (decimal, DMS, MGRS, UTM/UPS, WKT) can be successfully copied to the clipboard and pasted accurately elsewhere. 

Verify that MGRS is not an option when clicking at the very top and bottom of the maps (by the poles). 

#### What are the relevant tickets?
[DDF-4317](https://codice.atlassian.net/browse/DDF-4317)

#### Screenshots
Do we think we need to move all the copy/paste options to a sub-menu? The top level would be **Copy coordinates as** and the options would unfurl to the side. Not sure if this is cluttered. The settings for the screenshots below include a `Theme` -> `Zoom Percentage` of `100%` and a `Spacing` of `Comfortable`. 

<img width="824" alt="screen shot 2018-11-12 at 6 52 02 pm" src="https://user-images.githubusercontent.com/7215534/48386305-b4fd5780-e6ae-11e8-8209-fcbfa8a85ca0.png">

<img width="828" alt="screen shot 2018-11-12 at 6 52 18 pm" src="https://user-images.githubusercontent.com/7215534/48386315-b9c20b80-e6ae-11e8-9252-e292925fab9a.png">

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
